config/check_gettid.m4

```diff
@@ -0,0 +1,16 @@
+#CHECK_GETTID([action-if-found], [action-if-not-found])
+# --------------------------------------------------------
+# check if SYS_gettid is available.
+AC_DEFUN([CHECK_GETTID], [
+    AC_MSG_CHECKING([SYS_gettid syscall definitions])
+    AC_LANG_PUSH([C])
+    AC_COMPILE_IFELSE([
+       AC_LANG_SOURCE([[#include <sys/syscall.h>]],
+                      [[#define _GNU_SOURCE]],
+                      [[long tid = SYS_gettid;]])],
```

Comment:
I think AC_LANG_SOURCE only takes one argument?  I suspect the other arguments are getting ignored.

It looks like ```_GNU_SOURCE``` needs to move up so it's defined before including ```sys/syscall.h```.  You could put an ```#ifndef __APPLE__``` in there to force this test to always pass on Mac.

Reply:
I verified that the second and third arguments are getting ignored, so this should be updated to:
```
        AC_LANG_SOURCE([[
#ifndef __APPLE__
#define _GNU_SOURCE
#include <sys/syscall.h>
long tid = SYS_gettid;
#endif
]])
```